### PR TITLE
Add opusfile recipe

### PR DESCRIPTION
**Pull request by asafkahlon**
Signed-off-by: Asaf Kahlon <asafka7@gmail.com>

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/opusfile/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://ftp.osuosl.org/pub/xiph/releases/opus/{{ name }}-{{ version }}.tar.gz
+  sha256: 74ce9b6cf4da103133e7b5c95df810ceb7195471e1162ed57af415fabf5603bf
+  patches:
+    - 0001-Disable-GNU_SOURCE.patch
```
Comment (marcelotrevisani) [inline recipes/opusfile/meta.yaml]: Could you please add a comment explaining why this patch is needed?

File: recipes/opusfile/build.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+./configure --prefix=${PREFIX}
+make
```
Comment (marcelotrevisani) [inline recipes/opusfile/build.sh]: maybe 
```suggestion
make -j${CPU_COUNT}
```

File: recipes/opusfile/meta.yaml (hunk 1 above)
Comment (asafkahlon) [reply recipes/opusfile/meta.yaml]: I just saw the `fmilib` package has the same issue, so I added a reference the Github issue as they did.

**Comment by scopatz** [comment]
LGTM

**Comment by asafkahlon** [comment]
@conda-forge/staged-recipes Is there anything else needed here?

**Comment by npavlovikj** [comment]
Thank you for contributing @asafkahlon 🎉

**Pull request merged**